src/pool/srv_pool.c

```diff
@@ -4381,3 +4382,32 @@ ds_pool_child_map_refresh_async(struct ds_pool_child *dpc)
 			    0, 0, NULL);
 	return rc;
 }
+
+
+int ds_pool_prop_fetch(struct ds_pool *pool, unsigned int bits,
+		       daos_prop_t **prop_out)
+{
+	struct pool_svc	*svc;
+	struct rdb_tx	tx;
+	int		rc;
+
+	rc = pool_svc_lookup_leader(pool->sp_uuid, &svc, NULL);
+	if (rc != 0)
+		return rc;
+
+	rc = rdb_tx_begin(svc->ps_rsvc.s_db, svc->ps_rsvc.s_term, &tx);
+	if (rc != 0)
+		D_GOTO(out_svc, rc);
+
+	/* read optional properties */
+	ABT_rwlock_rdlock(svc->ps_lock);
+	rc = pool_prop_read(&tx, svc, bits, prop_out);
+	ABT_rwlock_unlock(svc->ps_lock);
+	if (rc != 0)
+		D_GOTO(out_tx, rc);
+out_tx:
```

Comment:
[Optional] Unnecessary?

Reply:
ah, sorry, copy paste failure.  I will clean it up in the later patch.